indico/modules/rb_new/operations/bookings.py

```diff
@@ -209,3 +214,44 @@ def get_booking_occurrences(booking):
     date_range = sorted(set(cand.start_dt.date() for cand in booking.occurrences))
     occurrences = group_by_occurrence_date(booking.occurrences)
     return date_range, occurrences
+
+
+def check_room_available(room, start_dt, end_dt):
+    occurrences = get_existing_room_occurrences(room, start_dt, end_dt, allow_overlapping=True)
+    prebookings = [occ for occ in occurrences if not occ.reservation.is_accepted]
+    bookings = [occ for occ in occurrences if occ.reservation.is_accepted]
+    unbookable_hours = get_rooms_unbookable_hours([room])
+    hours_overlap = [hours for hours in unbookable_hours
+                     if overlaps((start_dt.time(), end_dt.time()), (hours.start_time, hours.end_time))]
+    nonbookable_periods = get_rooms_nonbookable_periods([room], start_dt, end_dt)
+    blockings = get_rooms_blockings([room], start_dt, end_dt).get(room.id, [])
+    blocked_for_user = [blocking for blocking in blockings
+                        if not blocking.blocking.can_be_overridden(session.user, room)]
+    user_bookings = [booking for booking in bookings if booking.reservation.booked_for_id == session.user.id]
```

Comment:
what about using `any(...)` instead of `[...]` here instead of converting to `bool()` below? probably there are a few more cases where we could do that